redis-cloud-aws-ts/README.md

```diff
@@ -0,0 +1,57 @@
+# Redis Enterprise Cloud AWS Example
+
+This example contains code that creates a [Redis Enterprise Cloud](https://app.redislabs.com/) cluster and connects it to an AWS VPC via VPC Peering. This stack also includes an EC2 instance with SSM Session Manager that can run Redis commands to verify connectivity to the Redis cluster.
+
+![Architecture diagram of an AWS VPC across 3 availability zones, connected to Redis Enterprise Cloud via VPC peering](images/architecture.png)
+
+## Prerequisites
+
+1. [Install Pulumi](https://www.pulumi.com/docs/get-started/install/)
+1. [Configure the Redis Cloud provider](https://www.pulumi.com/registry/packages/rediscloud/installation-configuration/)
+1. Ensure you have a payment method specified in the Redis Enterprise Cloud console.
+1. [Configure the AWS Classic provider](https://www.pulumi.com/registry/packages/aws/installation-configuration/)
+1. [Install NodeJS](https://nodejs.org/en/download)
+
+## Deploy the app
+
+1. Set configuration values for your Redis Cloud payment information:
+
+    ```bash
+    pulumi config set cardType <YOUR-CARD-TYPE> # e.g. Visa (capitalization matters)
+    pulumi config set lastFourNumbers 1234
+    ```
+
+1. If you are using an availability zone other than `us-east-1`, you will need to set `preferredAz`, e.g. if you're using `us-west-2`:
+
+    ```bash
+    pulumi config set aws:region us-west-2 # Not necessary if your default AWS region is e.g. us-west-2
+    pulumi config set preferredAz usw2-az1 # This step is necessary is your default or selected region is not us-east-1
+    ```
+
+1. Deploy the architecture:
+
+    ```bash
+    pulumi up
```

Comment:
First guess at what went wrong: "Visa" is case-sensitive.
Second guess at what went wrong: You have to use a Flexible account. Can't use Fixed as it's a limitation of the provider.

Reply:
^ Okay, maybe it's worthwhile adding this note as a remainder for others.

Reply:
I have this called out already in the README. Is there anyplace else I could put it that would make it more obvious? Unfortunately, I can't really control the behavior of the provider.

The provider also calls this out implicitly: https://www.pulumi.com/registry/packages/rediscloud/api-docs/getpaymentmethod/#cardtype_nodejs

Reply:
I think the docs in the Registry are highly ambiguous; just because it's capitalized there does not, to me, indicate that it _must_ be capitalized. It might be overkill, but can we add a code comment around line 23 or 24 mentioning that the configuration value must be capitalized or it will generate an error?

Reply:
Done.